zio-http/src/main/scala/zio/http/service/SSLConfig.scala

```diff
@@ -0,0 +1,57 @@
+package zio.http.service
+
+import zio.http.service.SSLConfig._
+
+case class SSLConfig(behaviour: SSLHttpBehaviour, data: SSLData, provider: SSLProvider)
+
+object SSLConfig {
+
+  def apply(data: SSLData): SSLConfig =
+    new SSLConfig(SSLHttpBehaviour.Redirect, data, SSLProvider.JDK)
+
+  def fromFile(certPath: String, keyPath: String): SSLConfig =
+    new SSLConfig(SSLHttpBehaviour.Redirect, SSLData.FromFile(certPath, keyPath), SSLProvider.JDK)
+
+  def fromFile(behaviour: SSLHttpBehaviour, certPath: String, keyPath: String): SSLConfig =
+    new SSLConfig(behaviour, SSLData.FromFile(certPath, keyPath), SSLProvider.JDK)
+
+  def fromResource(certPath: String, keyPath: String): SSLConfig =
+    new SSLConfig(SSLHttpBehaviour.Redirect, SSLData.FromResource(certPath, keyPath), SSLProvider.JDK)
+
+  def fromResource(behaviour: SSLHttpBehaviour, certPath: String, keyPath: String): SSLConfig =
+    new SSLConfig(behaviour, SSLData.FromResource(certPath, keyPath), SSLProvider.JDK)
+
+  def generate: SSLConfig =
+    new SSLConfig(SSLHttpBehaviour.Redirect, SSLData.Generate, SSLProvider.JDK)
+
+  def generate(behaviour: SSLHttpBehaviour): SSLConfig =
+    new SSLConfig(behaviour, SSLData.Generate, SSLProvider.JDK)
+
+  sealed trait SSLHttpBehaviour
```

Comment:
Because these helper classes are all located inside `SSLConfig`, we can use a shorter naming convention, e.g. `HttpBehavior` instead of `SSLHttpBehavior`, `Data` instead of `SSLData`.